### ISSUE TITLE
Failed to replace Qdisc for lxcfc6dee5519e6: Link not found

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#6059]
Link:
https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1802/execution/node/98/log/?consoleFull

```
15:38:06 STEP: Waiting for all etcd-operator pods are ready
15:38:54 STEP: Applying deployments
15:38:57 STEP: Waiting for pods to be ready
15:44:10 === Test Finished at 2018-10-24T15:43:59Z====
15:44:18 ===================== TEST FAILED =====================
15:44:25 prometheus-core-5c9fbf5cc6-j59nk => none
15:44:25 deathstar-7bbc6d774d-8kz26 => none
15:44:25 etcd-operator-8b5cbb6d6-znr96 => none
15:44:25 deathstar-7bbc6d774d-79ljj => none
15:44:25 spaceship-6db9c5df76-msf2q => none
15:44:25 cilium-etcd-ztslk8vpf2 => none
15:44:25 kube-dns-6f4c8dbcd4-2bckf => none
15:44:25 cilium-etcd-cbhx49xbjm => none
15:44:25 cilium-etcd-lztngq7fbr => none
15:44:25 microscope => none
15:44:25 spaceship-6db9c5df76-9k22d => none
15:44:25 spaceship-6db9c5df76-k92dd => none
15:44:25 spaceship-6db9c5df76-wvp6c => none
15:44:25 xwing-f6d5fc5-vlhcv => none
15:44:30 <Checks>
15:44:30 ⚠️  Found a 'JoinEP: ' in logs
15:44:30 Number of 'context deadline exceeded' in logs: 4
15:44:30 ⚠️  Number of 'level=error' in logs: 15
15:44:30 ⚠️  Number of 'level=warning' in logs: 85
15:44:30 Number of 'Cilium API handler panicked' in logs: 0
15:44:30 ⚠️  Number of 'Goroutine took lock for more than' in logs: 542
15:44:30 Cilium pods: [cilium-7tmcj cilium-r4dcg]
15:44:30 Netpols loaded: 
15:44:30 CiliumNetworkPolicies loaded: 
15:44:30 Endpoint Policy Enforcement:
15:44:30      prometheus-core-5c9fbf5cc6-j59nk   =>   none
15:44:30      deathstar-7bbc6d774d-8kz26         =>   none
15:44:30      etcd-operator-8b5cbb6d6-znr96      =>   none
15:44:30      deathstar-7bbc6d774d-79ljj         =>   none
15:44:30      spaceship-6db9c5df76-msf2q         =>   none
15:44:30      cilium-etcd-ztslk8vpf2             =>   none
15:44:30      kube-dns-6f4c8dbcd4-2bckf          =>   none
15:44:30      cilium-etcd-cbhx49xbjm             =>   none
15:44:30      cilium-etcd-lztngq7fbr             =>   none
15:44:30      microscope                         =>   none
15:44:30      spaceship-6db9c5df76-9k22d         =>   none
15:44:30      spaceship-6db9c5df76-k92dd         =>   none
15:44:30      spaceship-6db9c5df76-wvp6c         =>   none
15:44:30      xwing-f6d5fc5-vlhcv                =>   none
15:44:30 
15:44:30 </Checks>
15:44:30 
15:44:30 [[ATTACHMENT|3410ba8d_K8sDemosTest_Tests_Star_Wars_Demo.zip]]
15:44:30 
15:44:30 • Failure [1631.186 seconds]
15:44:30 K8sDemosTest
15:44:30 /home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:351
15:44:30   Tests Star Wars Demo [It]
15:44:30   /home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:383
15:44:30 
15:44:30   Pods are not ready after timeout
15:44:30   Expected
15:44:30       <*errors.errorString | 0xc4201dc870>: {
15:44:30           s: "Timeout reached: timed out waiting for pods with filter  to be ready",
15:44:30       }
15:44:30   to be nil
15:44:30 
15:44:30   /home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/k8sT/demos.go:118
```

From `cilium-test.log`:
```
2018-10-24T15:42:44.573472079Z level=warning msg="JoinEP: Failed to load program" containerID=3ef3d84dc0 datapathPolicyRevision=0 desiredPolicyRevision=3 endpointID=37832 error="Failed to replace Qdisc for lxcfc6dee5519e6: Link not found" file-path=37832_next/bpf_lxc.o ipv4=10.10.0.52 ipv6="f00d::a0a:0:0:93c8" k8sPodName=default/spaceship-6db9c5df76-k92dd subsys=datapath-loader veth=lxcfc6dee5519e6
2018-10-24T15:42:44.573493534Z level=info msg="Recompiled endpoint BPF program" BPFCompilationTime=3m28.91959295s containerID=3ef3d84dc0 datapathPolicyRevision=0 desiredPolicyRevision=3 endpointID=37832 error="Failed to replace Qdisc for lxcfc6dee5519e6: Link not found" ipv4=10.10.0.52 ipv6="f00d::a0a:0:0:93c8" k8sPodName=default/spaceship-6db9c5df76-k92dd subsys=endpoint
2018-10-24T15:42:44.573615477Z level=error msg="Restoring endpoint state after BPF regeneration failed" containerID=3ef3d84dc0 datapathPolicyRevision=0 desiredPolicyRevision=3 endpointID=37832 ipv4=10.10.0.52 ipv6="f00d::a0a:0:0:93c8" k8sPodName=default/spaceship-6db9c5df76-k92dd subsys=endpoint
```

We may need to check upon endpoint restore that the link device for an endpoint still exists, it seems like perhaps the device disappeared while Cilium was down/restarting.

Unfortunately for this one, we're missing most of the bugtool output from the zip (@eloycoto , any ideas why?)

[3410ba8d_K8sDemosTest_Tests_Star_Wars_Demo (1).zip](https://github.com/cilium/cilium/files/2516778/3410ba8d_K8sDemosTest_Tests_Star_Wars_Demo.1.zip)

**Comment by joestringer** [comment]
Similar issue in the same CI failure, test is `K8sUpdates Tests upgrade and downgrade from a Cilium stable image to master`:

`pod-kube-system-cilium-dqrhl-cilium-agent.log`:
```
2018-10-24T16:07:46.468474736Z level=warning msg="Command execution failed" cmd="/var/lib/cilium/bpf/join_ep.sh /var/lib/cilium/bpf /var/run/cilium/state /var/run/cilium/state/57558_next lxcac714 false 57558" containerID=ac71451969 endpointID=57558 error="exit status 1" ipv4=10.10.0.245 ipv6="f00d::a0a:0:0:e0d6" k8sPodName=/ policyRevision=0
2018-10-24T16:07:46.468521873Z level=warning msg="Join EP id=/var/run/cilium/state/57558_next ifname=lxcac714" subsys=endpoint
2018-10-24T16:07:46.468538572Z level=warning msg="'probe' is not a recognized processor for this target (ignoring processor)" subsys=endpoint
2018-10-24T16:07:46.468554015Z level=warning msg="'probe' is not a recognized processor for this target (ignoring processor)" subsys=endpoint
2018-10-24T16:07:46.468568768Z level=warning msg="Cannot find device \"lxcac714\"" subsys=endpoint
2018-10-24T16:07:46.468584212Z level=warning msg="Cannot find device \"lxcac714\"" subsys=endpoint
2018-10-24T16:07:46.468604902Z level=info msg="Recompiled endpoint BPF program" BPFCompilationTime=1m59.728423699s containerID=ac71451969 endpointID=57558 error="error: \"exit status 1\" command output: \"Join EP id=/var/run/cilium/state/57558_next ifname=lxcac714\\n'probe' is not a recognized processor for this target (ignoring processor)\\n'probe' is not a recognized processor for this target (ignoring processor)\\nCannot find device \\\"lxcac714\\\"\\nCannot find device \\\"lxcac714\\\"\\n\"" ipv4=10.10.0.245 ipv6="f00d::a0a:0:0:e0d6" k8sPodName=default/app2-75f77ffdbf-bmnvs policyRevision=0
2018-10-24T16:07:46.468631357Z level=error msg="Restoring endpoint state after BPF regeneration failed" containerID=ac71451969 endpointID=57558 ipv4=10.10.0.245 ipv6="f00d::a0a:0:0:e0d6" k8sPodName=default/app2-75f77ffdbf-bmnvs policyRevision=0
2018-10-24T16:07:46.468648904Z level=debug msg="Reverting proxy redirect removals" containerID=ac71451969 endpointID=57558 ipv4=10.10.0.245 ipv6="f00d::a0a:0:0:e0d6" k8sPodName=default/app2-75f77ffdbf-bmnvs policyRevision=0
2018-10-24T16:07:46.46866499Z level=debug msg="Finished reverting proxy redirect removals" containerID=ac71451969 endpointID=57558 ipv4=10.10.0.245 ipv6="f00d::a0a:0:0:e0d6" k8sPodName=default/app2-75f77ffdbf-bmnvs policyRevision=0
2018-10-24T16:07:46.46868156Z level=debug msg="Reverting proxy redirect additions" containerID=ac71451969 endpointID=57558 ipv4=10.10.0.245 ipv6="f00d::a0a:0:0:e0d6" k8sPodName=default/app2-75f77ffdbf-bmnvs policyRevision=0
2018-10-24T16:07:46.468697968Z level=debug msg="Finished reverting proxy redirect additions" containerID=ac71451969 endpointID=57558 ipv4=10.10.0.245 ipv6="f00d::a0a:0:0:e0d6" k8sPodName=default/app2-75f77ffdbf-bmnvs policyRevision=0
2018-10-24T16:07:46.468713808Z level=debug msg="Reverting xDS network policy update" subsys=envoy-manager
2018-10-24T16:07:46.468727315Z level=debug msg="preparing new cache transaction: upserting 0 entries, deleting 1 entries" subsys=xds xdsTypeURL=type.googleapis.com/cilium.NetworkPolicy xdsVersionInfo=94
2018-10-24T16:07:46.468743383Z level=debug msg="deleting resource from cache" subsys=xds xdsResourceName="f00d::a0a:0:0:e0d6" xdsTypeURL=type.googleapis.com/cilium.NetworkPolicy xdsVersionInfo=94
2018-10-24T16:07:46.468758683Z level=debug msg="committing cache transaction and notifying of new version" subsys=xds xdsTypeURL=type.googleapis.com/cilium.NetworkPolicy xdsVersionInfo=94
2018-10-24T16:07:46.46877558Z level=debug msg="preparing new cache transaction: upserting 0 entries, deleting 1 entries" subsys=xds xdsTypeURL=type.googleapis.com/cilium.NetworkPolicy xdsVersionInfo=95
2018-10-24T16:07:46.468806758Z level=debug msg="deleting resource from cache" subsys=xds xdsResourceName=10.10.0.245 xdsTypeURL=type.googleapis.com/cilium.NetworkPolicy xdsVersionInfo=95
2018-10-24T16:07:46.468823423Z level=debug msg="committing cache transaction and notifying of new version" subsys=xds xdsTypeURL=type.googleapis.com/cilium.NetworkPolicy xdsVersionInfo=95
2018-10-24T16:07:46.468836688Z level=debug msg="Finished reverting xDS network policy update" subsys=envoy-manager
2018-10-24T16:07:46.468851035Z level=error msg="Finished restoring endpoint state after BPF regeneration failed" containerID=ac71451969 endpointID=57558 ipv4=10.10.0.245 ipv6="f00d::a0a:0:0:e0d6" k8sPodName=default/app2-75f77ffdbf-bmnvs policyRevision=0
```

[541d8bab_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master (1).zip](https://github.com/cilium/cilium/files/2516796/541d8bab_K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master.1.zip)

**Comment by joestringer** [comment]
It is likely that this is a long-running issue spanning back as far as 1.2 or earlier, but was masked by the fact that we ignored all errors while replacing Qdiscs. On master (v1.4 cycle), we may be observing it again because https://github.com/cilium/cilium/pull/5948 stops hiding such issues.

**Comment by joestringer** [comment]
I suspect that if the endpoint being restored ends up being etcd, then this could cause Cilium to remain not ready for quite a long time.

**Comment by joestringer** [comment]
This failure may also appear as:

```
15:54:58 • Failure in Spec Teardown (AfterEach) [2258.771 seconds]
15:54:58 K8sPolicyTest
15:54:58 /home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:351
15:54:58   Basic Test
15:54:58   /home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:351
15:54:58     checks all kind of Kubernetes policies [AfterEach]
15:54:58     /home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:383
15:54:58 
15:54:58     Found a 'JoinEP: ' in Cilium Logs
15:54:58 
15:54:58     /home/jenkins/workspace/cilium-ginkgo_cilium_master-QVKDPVJ2A6E272IRUX7X6DCB5FBLCQDKRTOMHPZS7ZECX3RM5C3A/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:474
```

https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1802/execution/node/99/log/?consoleFull

**Comment by ianvernon** [comment]
This is still occurring intermittently in master:

Build link: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/1983/testReport/junit/k8s-1/12/K8sTunnelTest_VXLan_Check_VXLAN_mode/

Failure Output:
```
Stacktrace

/home/jenkins/workspace/cilium-ginkgo_cilium_master/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:313
Found a 'JoinEP: ' in Cilium Logs
/home/jenkins/workspace/cilium-ginkgo_cilium_master/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:474
Standard Output

⚠️  Found a 'JoinEP: ' in logs
Number of 'context deadline exceeded' in logs: 2
Number of 'level=error' in logs: 5
⚠️  Number of 'level=warning' in logs: 49
Number of 'Cilium API handler panicked' in logs: 0
⚠️  Number of 'Goroutine took lock for more than' in logs: 48
Cilium pods: [cilium-4t5g6 cilium-8nwcd]
Netpols loaded: 
CiliumNetworkPolicies loaded: 
Endpoint Policy Enforcement:
     testclient-qfx7z                   =>   none
     cilium-etcd-vrdqz2mzkf             =>   none
     cilium-etcd-ztg7jk6btr             =>   none
     coredns-779ffd89bd-gw8bg           =>   none
     etcd-operator-547c5c7f84-8mqgw     =>   none
     prometheus-core-5888d94764-hbq2v   =>   none
     testclient-wqm9q                   =>   none
     testds-8rb7r                       =>   none
     testds-bq9zj                       =>   none
     cilium-etcd-plgwjgmjsh             =>   none
Cilium agent 'cilium-4t5g6': Status: Ok  Health: Ok Nodes "k8s2 k8s1" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 35 Failed 0
Cilium agent 'cilium-8nwcd': Status: Ok  Health: Ok Nodes "k8s1 k8s2" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 30 Failed 0

Standard Error

STEP: Installing DNS Deployment
STEP: Deploying etcd-operator
STEP: Installing Cilium
STEP: Performing Cilium preflight check
STEP: Checking that no controllers in Cilium have failed
STEP: Checking that cilium-health status is healthy
STEP: Checking that Kubernetes services are correctly plumbed by Cilium
STEP: Checking that the kubernetes service is correctly plumbed by Cilium
STEP: Waiting for all etcd-operator pods are ready
STEP: Checking that the kubernetes service is correctly plumbed by Cilium
STEP: Making sure all endpoints are in ready state
STEP: Checking that BPF tunnels are in place
STEP: Checking that BPF tunnels are working correctly
=== Test Finished at 2018-12-01T18:10:45Z====
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
Exitcode: 0 
Stdout:
 	 NAMESPACE     NAME                                    READY   STATUS    RESTARTS   AGE     IP              NODE   NOMINATED NODE
	 default       testclient-qfx7z                        1/1     Running   0          3m39s   10.10.1.103     k8s2   <none>
	 default       testclient-wqm9q                        1/1     Running   0          3m34s   10.10.0.101     k8s1   <none>
	 default       testds-8rb7r                            1/1     Running   0          3m39s   10.10.1.170     k8s2   <none>
	 default       testds-bq9zj                            1/1     Running   0          3m34s   10.10.0.179     k8s1   <none>
	 kube-system   cilium-4t5g6                            1/1     Running   2          3m43s   192.168.36.12   k8s2   <none>
	 kube-system   cilium-8nwcd                            1/1     Running   2          3m43s   192.168.36.11   k8s1   <none>
	 kube-system   cilium-etcd-operator-66b9b8c4d9-m9qc9   1/1     Running   0          3m46s   192.168.36.12   k8s2   <none>
	 kube-system   cilium-etcd-plgwjgmjsh                  1/1     Running   0          91s     10.10.1.70      k8s2   <none>
	 kube-system   cilium-etcd-vrdqz2mzkf                  1/1     Running   0          3m7s    10.10.0.156     k8s1   <none>
	 kube-system   cilium-etcd-ztg7jk6btr                  1/1     Running   0          51s     10.10.0.157     k8s1   <none>
	 kube-system   coredns-779ffd89bd-gw8bg                1/1     Running   0          7m27s   10.10.1.47      k8s2   <none>
	 kube-system   etcd-k8s1                               1/1     Running   0          6m26s   192.168.36.11   k8s1   <none>
	 kube-system   etcd-operator-547c5c7f84-8mqgw          1/1     Running   0          3m28s   10.10.1.186     k8s2   <none>
	 kube-system   kube-apiserver-k8s1                     1/1     Running   0          6m29s   192.168.36.11   k8s1   <none>
	 kube-system   kube-controller-manager-k8s1            1/1     Running   0          6m25s   192.168.36.11   k8s1   <none>
	 kube-system   kube-proxy-6gcpp                        1/1     Running   0          4m9s    192.168.36.12   k8s2   <none>
	 kube-system   kube-proxy-gsvpz                        1/1     Running   0          7m26s   192.168.36.11   k8s1   <none>
	 kube-system   kube-scheduler-k8s1                     1/1     Running   0          6m41s   192.168.36.11   k8s1   <none>
	 monitoring    prometheus-core-5888d94764-hbq2v        1/1     Running   0          3m46s   10.10.1.124     k8s2   <none>
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-4t5g6 -- cilium bpf tunnel list
Exitcode: 0 
Stdout:
 	 TUNNEL            VALUE           
	 10.10.0.0         192.168.36.11   
	 f00d::a0a:0:0:0   192.168.36.11   
	 
Stderr:
 	 level=info msg="Detected MTU 1500" subsys=mtu
	 

cmd: kubectl exec -n kube-system cilium-4t5g6 -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                
	 5762       Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0a:100:0:b552   10.10.1.70    ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                          
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.cilium/app=etcd-operator                                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 24016      Disabled           Disabled          25904      k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:f86b   10.10.1.103   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.kubernetes.pod.namespace=default                                                               
	                                                            k8s:zgroup=testDSClient                                                                               
	 43513      Disabled           Disabled          2664       k8s:app=prometheus                                       f00d::a0a:100:0:3420   10.10.1.124   ready   
	                                                            k8s:component=core                                                                                    
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=prometheus-k8s                                                
	                                                            k8s:io.kubernetes.pod.namespace=monitoring                                                            
	 47708      Disabled           Disabled          100        k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:3f4b   10.10.1.186   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa                                                
	                                                            k8s:io.cilium/app=etcd-operator                                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 51018      Disabled           Disabled          104        k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:896d   10.10.1.47    ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=coredns                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	                                                            k8s:k8s-app=kube-dns                                                                                  
	 64547      Disabled           Disabled          13700      k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:4e49   10.10.1.170   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.kubernetes.pod.namespace=default                                                               
	                                                            k8s:zgroup=testDS                                                                                     
	 65143      Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:e7b8   10.10.1.145   ready   
	 
Stderr:
 	 level=info msg="Detected MTU 1500" subsys=mtu
	 

cmd: kubectl exec -n kube-system cilium-8nwcd -- cilium bpf tunnel list
Exitcode: 0 
Stdout:
 	 TUNNEL              VALUE           
	 f00d::a0a:100:0:0   192.168.36.12   
	 10.10.1.0           192.168.36.12   
	 
Stderr:
 	 level=info msg="Detected MTU 1500" subsys=mtu
	 

cmd: kubectl exec -n kube-system cilium-8nwcd -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                       IPv6                 IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                       
	 2967       Disabled           Disabled          4          reserved:health                                   f00d::a0a:0:0:14f0   10.10.0.244   ready   
	 19199      Disabled           Disabled          13700      k8s:io.cilium.k8s.policy.cluster=default          f00d::a0a:0:0:1a97   10.10.0.179   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
	                                                            k8s:io.kubernetes.pod.namespace=default                                                      
	                                                            k8s:zgroup=testDS                                                                            
	 25887      Disabled           Disabled          25904      k8s:io.cilium.k8s.policy.cluster=default          f00d::a0a:0:0:19dd   10.10.0.101   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
	                                                            k8s:io.kubernetes.pod.namespace=default                                                      
	                                                            k8s:zgroup=testDSClient                                                                      
	 40856      Disabled           Disabled          101        k8s:app=etcd                                      f00d::a0a:0:0:26d6   10.10.0.157   ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                 
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                     
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
	                                                            k8s:io.cilium/app=etcd-operator                                                              
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                  
	 41242      Disabled           Disabled          101        k8s:app=etcd                                      f00d::a0a:0:0:24a9   10.10.0.156   ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                 
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                     
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
	                                                            k8s:io.cilium/app=etcd-operator                                                              
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                  
	 
Stderr:
 	 level=info msg="Detected MTU 1500" subsys=mtu
```

Logs: 
[e7cf1244_K8sTunnelTest_VXLan_Check_VXLAN_mode.zip](https://github.com/cilium/cilium/files/2640324/e7cf1244_K8sTunnelTest_VXLan_Check_VXLAN_mode.zip)


From Cilium logs:

```
2018-12-01T18:09:11.164273343Z level=warning msg="JoinEP: Failed to load program" containerID=ca6df59ee0 datapathPolicyRevision=0 desiredPolicyRevision=1 endpointID=0 error="Failed to replace Qdisc for lxc1e1c9ad5782a: Link not found" file-path=50588_next/bpf_lxc.o ipv4=10.10.1.210 ipv6="f00d::a0a:100:0:a25e" k8sPodName=default/testclient-qfx7z subsys=datapath-loader veth=lxc1e1c9ad5782a
2018-12-01T18:09:11.954207641Z level=warning msg="JoinEP: Failed to load program" containerID=e3d6417e93 datapathPolicyRevision=0 desiredPolicyRevision=2 endpointID=0 error="Failed to replace Qdisc for lxccc474263ae6b: Link not found" file-path=53592_next/bpf_lxc.o ipv4=10.10.1.150 ipv6="f00d::a0a:100:0:2f8d" k8sPodName=monitoring/prometheus-core-5888d94764-hbq2v subsys=datapath-loader veth=lxccc474263ae6b
2018-12-01T18:09:12.702491957Z level=warning msg="JoinEP: Failed to load program" containerID=4be455fca2 datapathPolicyRevision=0 desiredPolicyRevision=3 endpointID=0 error="Failed to replace Qdisc for lxc132b7d2720b3: Link not found" file-path=62870_next/bpf_lxc.o ipv4=10.10.1.165 ipv6="f00d::a0a:100:0:a7e6" k8sPodName=default/testds-8rb7r subsys=datapath-loader veth=lxc132b7d2720b3
2018-12-01T18:09:13.413281369Z level=warning msg="JoinEP: Failed to load program" containerID=ca6df59ee0 datapathPolicyRevision=0 desiredPolicyRevision=3 endpointID=0 error="Failed to replace Qdisc for lxc1e1c9ad5782a: Link not found" file-path=50588_next/bpf_lxc.o ipv4=10.10.1.210 ipv6="f00d::a0a:100:0:a25e" k8sPodName=default/testclient-qfx7z subsys=datapath-loader veth=lxc1e1c9ad5782a
2018-12-01T18:09:14.289329082Z level=warning msg="JoinEP: Failed to load program" containerID=e3d6417e93 datapathPolicyRevision=0 desiredPolicyRevision=3 endpointID=0 error="Failed to replace Qdisc for lxccc474263ae6b: Link not found" file-path=53592_next/bpf_lxc.o ipv4=10.10.1.150 ipv6="f00d::a0a:100:0:2f8d" k8sPodName=monitoring/prometheus-core-5888d94764-hbq2v subsys=datapath-loader veth=lxccc474263ae6b
2018-12-01T18:09:18.893772533Z level=warning msg="JoinEP: Failed to load program" containerID=4be455fca2 datapathPolicyRevision=0 desiredPolicyRevision=3 endpointID=0 error="Failed to replace Qdisc for lxc132b7d2720b3: Link not found" file-path=62870_next/bpf_lxc.o ipv4=10.10.1.165 ipv6="f00d::a0a:100:0:a7e6" k8sPodName=default/testds-8rb7r subsys=datapath-loader veth=lxc132b7d2720b3
```

**Comment by brb** [comment]
Re-opening, as the error was observed on one of the CI runs of https://github.com/cilium/cilium/pull/6851 which postdates #6551.

* https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1578/testReport/junit/k8s-1/11/BDD_Test_k8s_1_11_and_1_12___K8sServicesTest_Checks_service_across_nodes_Checks_ClusterIP_Connectivity/
* [K8sServicesTest_Checks_service_across_nodes_Checks_ClusterIP_Connectivity.zip](https://github.com/cilium/cilium/files/2812993/K8sServicesTest_Checks_service_across_nodes_Checks_ClusterIP_Connectivity.zip)

**Comment by brb** [comment]
One more found JoinEP due to shutdown:

```
Feb 20 07:33:17 runtime cilium-agent[20229]: level=error msg="Failed to compile bpf_lxc.o: signal: terminated" compiler-pid=21087 linker-pid=21088 subsys=datapath-loader
Feb 20 07:33:17 runtime cilium-agent[20229]: level=warning msg="JoinEP: Failed to compile" containerID= datapathPolicyRevision=0 debug=false desiredPolicyRevision=21 endpointID=2779 error="Failed to compile bpf_lxc.o: signal: terminated" identity=4 ipv4=10.15.172.21 ipv6="f00d::a0f:0:0:f7a3" k8sPodName=/ params="&{Source:bpf_lxc.c Output:bpf_lxc.o OutputType:obj}" subsys=datapath-loader
Feb 20 07:33:17 runtime cilium-agent[20229]: level=info msg="Recompiled endpoint BPF program" BPFCompilationTime=726.48788ms containerID= datapathPolicyRevision=0 desiredPolicyRevision=21 endpointID=2779 error="Failed to compile bpf_lxc.o: signal: terminated" identity=4 ipv4=10.15.172.21 ipv6="f00d::a0f:0:0:f7a3" k8sPodName=/ subsys=endpoint
<..>
Feb 20 07:33:17 runtime cilium-agent[20229]: level=info msg="Shutting down... " subsys=daemon
```

https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/5958/testReport/junit/(root)/runtime/RuntimeKVStoreTest_Consul_KVStore/

**Comment by brb** [comment]
And more:

```
level=error msg="Failed to compile bpf_lxc.o: signal: killed" compiler-pid=21131 linker-pid=21132 subsys=datapath-loader
level=warning msg="JoinEP: Failed to compile" debug=false error="Failed to compile bpf_lxc.o: signal: killed" params="&{Source:bpf_lxc.c Output:bpf_lxc.o OutputType:obj}" subsys=testutils
```

https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/5953/testReport/junit/(root)/runtime/RuntimePolicyImportTests_Invalid_Policies/

**Comment by joestringer** [comment]
Are these latest failures just that we are reporting a warning and failing if there's warnings in the log? Maybe the log just needs to be a little more smart during shutdown.

**Comment by brb** [comment]
> Are these latest failures just that we are reporting a warning and failing if there's warnings in the log?

Yes, that's the case.

> Maybe the log just needs to be a little more smart during shutdown.

Either this, or we should relax the JoinEP check by ignoring ones logged due to termination.

**Comment by brb** [comment]
One more: https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/5972/testReport/junit/(root)/runtime/RuntimePolicyEnforcement_Policy_Enforcement_Default_Default_values/

**Comment by brb** [comment]
One more: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/2417/

**Comment by brb** [comment]
Fixed by #7203.
